basetest.pm

```diff
@@ -375,6 +375,20 @@ sub next_resultname {
     }
 }
 
+sub record_resultfile {
+    my ($self, $title, $output, $result) = @_;
+    my $filename = $self->next_resultname('txt');
```

Comment:
We use both of `next_resultname`'s arguments, so you probably should add an optional named argument for that.